x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_host_table.ts

```diff
@@ -7,19 +7,28 @@
 
 import { last } from 'lodash';
 import { useMemo } from 'react';
-import type { SnapshotNode, SnapshotNodeMetric } from '../../../../../common/http_api';
-import { HostMetics } from '../components/hosts_table_columns';
+import { SnapshotNode } from '../../../../../common/http_api';
+import { HostMetics, HostNodeRow } from '../components/hosts_table_columns';
 
-type MappedMetrics = Record<keyof HostMetics, SnapshotNodeMetric>;
+export type MappedMetrics = Record<keyof HostMetics, number | null | undefined>;
 
 export const useHostTable = (nodes: SnapshotNode[]) => {
-  const items: MappedMetrics[] = useMemo(() => {
-    return nodes.map(({ metrics, path }) => ({
-      ...last(path),
+  const items: HostNodeRow[] = useMemo(() => {
+    const valuesMapping: Record<keyof HostMetics, 'value' | 'avg' | 'max'> = {
```

Comment:
This is a smart solution :). If we don't need to make the changes on `hosts_table_column` we may not need this anymore - which would be even simpler

Reply:
I changed the columns because for the sorting we need the fields to have the value as a number or string - the value we want to use for sorting the table - so for example `memory: {avg: 1, value: 2, max: 3}` should be `memory: 1` because we show the `avg` in the table and for sorting the column the average value (`1`)  will be used ( similar to all the columns ). That's why I changed the structure there.